sgkit/utils.py

```diff
@@ -1,7 +1,14 @@
+from typing import Any, Set, Union
+
 import numpy as np
 
 
-def check_array_like(a, dtype=None, kind=None, ndim=None):
+def check_array_like(
+    a: Any,
+    dtype: Union[None, str, Set[str]] = None,
```

Comment:
Should we also allow `numpy.dtype`?

Reply:
Good point. The number of different ways of specifying types is quite long (see https://numpy.org/doc/stable/reference/arrays.dtypes.html#arrays-dtypes-constructing), so I think specifying Any is probably the best we can do. I noticed that xarray doesn't generally declare types for dtype arguments, and where it does they are Any.

Reply:
@tomwhite yea, I try to avoid `Any` for obvious reasons. I wonder if it would make sense to add a typing module within sgkit to start gradually defining these kind of types, so we could create an `DTypeType` (or sth like that) which right now could either be `Any` or more gradual `Union[Any, <other_types>]` (the extra types are just informative), neither way is type safe but at least it's a bit more future proof, when maybe in the future when we have a better way to define these types it would be just a matter of swapping the type definition?

Reply:
Sounds good. I added a `DType` alias for `Any`.